src/Symfony/Component/Messenger/Bridge/Redis/Transport/Connection.php

```diff
@@ -523,7 +523,20 @@ public function cleanup(): void
 
     public function getMessageCount(): int
     {
-        return $this->connection->xlen($this->stream);
+        $groups = $this->connection->xinfo('GROUPS', $this->stream);
+
+        if (false === $groups) {
+            return 0;
```

Comment:
Is this valid that we return `0` also when there could be messages in there? Is there a special exception we need to throw in that case. Return 0 sounds strange for the case where we are not able to count the messages.

Reply:
Replaced by a `LogicException` like in the `setup` method.

Reply:
@chalasr Do we need extend here something in the messenger component. It should behave like before when we are not able to count the messages. Will the LogicException be catched by messenger and behave so like before?

Reply:
@chalasr Do we need extend here something in the messenger component. It should behave like before when we are not able to count the messages. Will the LogicException be catched by messenger and behave so like before?